src/components/modals/FormStepSaveModal.js

```diff
@@ -114,9 +115,12 @@ const FormStepSaveModal = ({
       return;
     }
 
+    onSessionDestroyed();
     // redirect back to start page
     dispatch({type: 'SAVE_SUCCEEDED'});
     history.push('/');
+    // TODO: replace with a proper reset of the state instead of a page reload.
+    window.location.reload();
```

Comment:
can't we do this as part of this hook then? should just be a reducer action type so you can `dispatch({type: 'RESET'})` in the form step, part of the `onSessionDestroyed`, no?